### DISCUSSION TITLE
Custom sorting ports

### DISCUSSION THREAD

**Discussion by darekdajcz** [clientIO/joint#2218] [Q&A] [answered]
### Introduction

Is there any option to sort ports in my shape. Only way I discovered is to remove old ones and add new  sorted. But unfortunately there is an issue. While removing old ports it's also goes to removing links with it...

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS+

**Comment by kumilingus** [comment, marked as answer]
If you only change the order of the ports in the `ports/items` array, no link will be removed.

```js
el.addPort({ attrs: { circle: { fill: 'red' }}});
el.addPort({ attrs: { circle: { fill: 'green' }}});
el.addPort({ attrs: { circle: { fill: 'blue' }}});

const ports = el1.getPorts();
const portsNewOrder = [ports[2], ports[0], ports[1]];
el.prop('ports/items', portsNewOrder, { rewrite: true });
```
